hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/SstFilteringService.java

```diff
@@ -148,6 +148,7 @@ public BackgroundTaskResult call() throws Exception {
                   dbName, true);
           RocksDatabase db = rdbStore.getDb();
           db.deleteFilesNotMatchingPrefix(prefixPairs, filterFunction);
+          rdbStore.close();
```

Comment:
Use try-with-resources to ensure close happens even in face of exceptions.

```java
          try (RDBStore rdbStore = (RDBStore) OmMetadataManagerImpl
              .loadDB(ozoneManager.getConfiguration(), new File(snapshotDir),
                  dbName, true)) {
            RocksDatabase db = rdbStore.getDb();
            db.deleteFilesNotMatchingPrefix(prefixPairs, filterFunction);
          }
```

Reply:
Good catch. @DaveTeng0 Would you post another PR to address this?

Reply:
good catch! yeah i'll do that!